src/public/manageDomain.ts

```diff
@@ -0,0 +1,173 @@
+const envList = helper.getElement('.envList')
+const environmentVariables: EnvironmentItem[] = []
+const fullDomain = window.location.pathname.split('/').pop()
+document.getElementById('full-domain').innerText = fullDomain
+
+class EnvironmentItem {
+  name: string
+  value: string
+  isValid = false
+
+  private nameInputElement: HTMLInputElement
+  private valueInputElement: HTMLInputElement
+
+  private readonly IS_INVALID = 'is-invalid'
+  private readonly LETTER_NUMBER_REGEX = /^[A-Z0-9_]+$/g
+  private readonly ELEMENT_HTML = `
+    <div class="input-group">
+      <input 
+        type="text"
+        aria-label="Name"
+        class="form-control text-uppercase text-monospace name-input ${this.IS_INVALID}"
+        placeholder="NAME"
+      >
+      <div class="input-group-prepend input-group-append">
+        <span class="input-group-text">=</span>
+      </div>
+      <input
+        type="text"
+        aria-label="Value"
+        class="form-control text-monospace value-input"
+        placeholder="Value"
+      >
+      <div class="input-group-append">
+        <button
+          class="btn btn-outline-danger remove-button"
+          type="button"
+        >
+          Remove
+        </button>
+      </div>
+      <div class="invalid-feedback">
+        Name must contain only letters, number, or underscore.
+      </div>
+    </div>
+  `
+
+  constructor(name?: string, value?: string) {
+    this.name = name
+    this.value = value
+    this.createElement()
+    if (name && value) {
+      this.setInputValues()
```

Comment:
I don't really see a need to even call `this.setInputValues` in the constructor, do you?

Reply:
You can always set the default values in the constructor and then `setInputValues` should be called upon updates.

Reply:
Also, I would probably rename this to something like `updateInputValues`